service/src/main/java/com/codeforcommunity/processor/ProtectedUserProcessorImpl.java

```diff
@@ -30,12 +31,17 @@ public ProtectedUserProcessorImpl(DSLContext db, Emailer emailer) {
   }
 
   @Override
-  public void deleteUser(JWTData userData) {
+  public void deleteUser(JWTData userData, DeleteUserRequest deleteUserRequest) {
+    UsersRecord user = db.selectFrom(USERS).where(USERS.ID.eq(userData.getUserId())).fetchOne();
+
+    if (!Passwords.isExpectedPassword(deleteUserRequest.getPassword(), user.getPasswordHash())) {
+      throw new WrongPasswordException();
+    }
+
     int userId = userData.getUserId();
 
     db.deleteFrom(VERIFICATION_KEYS).where(VERIFICATION_KEYS.USER_ID.eq(userId)).executeAsync();
 
-    UsersRecord user = db.selectFrom(USERS).where(USERS.ID.eq(userId)).fetchOne();
     user.delete();
```

Comment:
This is unrelated to the change here but currently I think this would throw an exception in production. We need to a cascading delete where we also delete any rows in the database that have a foreign key relation to this user. Unfortunately, jOOQ doesn't have built in support for this right now so we should either manually delete rows that include a userID in them one-by-one or we could try running a raw SQL query that includes CASCADE. We should create a new ticket for this work so that we don't jam up this PR